src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs

```diff
@@ -698,13 +699,18 @@ public IActionResult BuildTransaction([FromBody] BuildTransactionRequest request
                     }
                 }
 
+                var (opReturnRawData, isValid) = request.OpReturnDataIsHex ? TryGetHexValue(request.OpReturnData) : (null, true);
+                if (!isValid)
+                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, "Hex OpReturn error.", "The OpReturnData is set as a hex value and could not be decoded");
+
                 var context = new TransactionBuildContext(this.network)
                 {
                     AccountReference = new WalletAccountReference(request.WalletName, request.AccountName),
                     TransactionFee = string.IsNullOrEmpty(request.FeeAmount) ? null : Money.Parse(request.FeeAmount),
                     MinConfirmations = request.AllowUnconfirmed ? 0 : 1,
                     Shuffle = request.ShuffleOutputs ?? true, // We shuffle transaction outputs by default as it's better for anonymity.
                     OpReturnData = request.OpReturnData,
+                    OpReturnRawData = opReturnRawData,
```

Comment:
Dont we need to also make changes inside the BuildTransaction for this to work?

Reply:
We are receiving the field/value on HEX and just converting it to bytes (binary) and use it as we are at this moment .. updating the OpRetunrRawData

Reply:
Right ok looks good to me this can go in